centos2ol.sh

```diff
@@ -272,6 +318,74 @@ rm -f "${reposdir}/switch-to-oraclelinux.repo"
 # At this point, the switch is completed.
 trap - ERR
 
+# When an additional enabled CentOS repository has a match with Oracle Linux
+#  then automatically enable the OL repository to ensure the RPM is maintained
+#
+# Create an associate array where the key is the CentOS reponame and the value
+#  contains the method of getting the content (Enable a repo or install an RPM)
+#  and the details of the repo or RPM
+case "$os_version" in
+    6*)
+        declare -A repositories=(
+            [base-debuginfo]="REPO https://oss.oracle.com/ol6/debuginfo/"
+            [updates]="REPO ol6_latest"
+        )
+        ;;
+    7*)
+        declare -A repositories=(
+            [base-debuginfo]="REPO https://oss.oracle.com/ol7/debuginfo/"
+            [updates]="REPO ol7_latest"
+            [centos-ceph-jewel]="RPM oracle-ceph-release-el7"
+            [centos-gluster41]="RPM oracle-gluster-release-el7"
+            [centos-gluster5]="RPM oracle-gluster-release-el7"
+            [centos-gluster46]="RPM oracle-gluster-release-el7"
+            [centos-nfs-ganesha30]="RPM oracle-gluster-release-el7"
+            [centos-ovirt42]="RPM oracle-ovirt-release-el7"
+            [centos-ovirt43]="RPM oracle-ovirt-release-el7"
+            [centos-sclo-sclo]="RPM oracle-softwarecollection-release-el7"
+            [centos-sclo-rh]="RPM oracle-softwarecollection-release-el7"
+        )
+        ;;
+    8*)
+        declare -A repositories=(
+            [AppStream]="REPO ol8_appstream"
+            [BaseOS]="REPO ol8_baseos_latest"
+            [HighAvailability]="REPO ol8_addons"
+            [PowerTools]="REPO ol8_codeready_builder"
+            [centos-release-nfs-ganesha28]="RPM oracle-gluster-release-el8"
+            [centos-gluster6-test]="RPM oracle-gluster-release-el8"
+            [centos-gluster7]="RPM oracle-gluster-release-el8"
+            [centos-gluster8]="RPM oracle-gluster-release-el8"
+        )
+        ;;
+esac
+
+# For each entry in the list, enable it
+for reponame in ${enabled_repos}; do
+    # action[0] will be REPO or RPM
+    # action[1] will be the repos details or the RPMs name
+    IFS=" " read -r -a action <<< "${repositories[${reponame}]}"
```

Comment:
This line is a bit hard to read but it's based on shellcheck best practices. I was happy with the older `action=(${repositories[${reponame}]})` I was using but I didn't want to introduce "issues"